pkg/controller/plan/adapter/vsphere/validator.go

```diff
@@ -0,0 +1,70 @@
+package vsphere
+
+import (
+	"fmt"
+	liberr "github.com/konveyor/controller/pkg/error"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1/ref"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web"
+	model "github.com/konveyor/forklift-controller/pkg/controller/provider/web/vsphere"
+)
+
+//
+// vSphere validator.
+type Validator struct {
+	plan      *api.Plan
+	inventory web.Client
+}
+
+//
+// Load.
+func (r *Validator) Load() (err error) {
+	r.inventory, err = web.NewClient(r.plan.Referenced.Provider.Source)
+	return
+}
+
+//
+// Validate that a VM's networks have been mapped.
+func (r *Validator) NetworksMapped(vmRef ref.Ref) (ok bool, err error) {
+	vm := &model.VM{}
+	pErr := r.inventory.Find(vm, vmRef)
```

Comment:
This _can_ assign `err` directly and it would be simpler.

Also, rather than creating a new error, you can wrap `err` with more context.
Eg:
```
err = liberr.Wrap(
    err,
    "VM not found."
   "vm",
  vmRef.String())
```
The example uses _structural_ logging instead of the sprintf.